Update Dockerfile

hTobi02 opened the pull request.

tgotwig commented on Dockerfile:
```diff
@@ -1,7 +1,18 @@
+COPY --from=mwader/static-ffmpeg:7.1.1 /ffmpeg /usr/local/bin/
+COPY --from=mwader/static-ffmpeg:7.1.1 /ffprobe /usr/local/bin/
+WORKDIR /app
+COPY --from=build /build/target/debug/vidmerger /usr/local/bin/
```
It should fetch it from a `release` folder rather than `debug`.

tgotwig commented on Dockerfile:
```diff
@@ -1,7 +1,18 @@
+    mv task /usr/local/bin/task
+WORKDIR /build
+COPY . /build
+RUN task
```
I think the `Taskfile.yml` needs to be updated a bit as well.

tgotwig reviewed (changes requested):
Hey @hTobi02 👋 Thanks for the PR! 🙌

Hmm somehow the image is much larger, from `docker images`:

```sh
REPOSITORY          TAG       IMAGE ID       CREATED          SIZE
vidmerger           latest    3a974e7d3415   45 minutes ago   364MB
tgotwig/vidmerger   latest    169420c6b1a2   13 months ago    113MB
```

---

`deps: [build_linux_x64]` can be removed from the Taskfile.

hTobi02 commented:
Hi @tgotwig,

The primary goal of this PR was to outsource the build process from the local environment and instead move it to a dedicated Docker stage. This way, the required dependencies no longer need to be installed locally, which makes it much easier to get started.

I’ve already addressed some of your comments and made the corresponding changes directly in this PR.

Regarding the image size: I was surprised by your figure of 113 MB. When I pull the current image from Docker Hub, I get about 164 MB. A significant part of the size comes from ffmpeg, which has grown considerably over time. While your image contains ffmpeg version 6.1.1, my new build already uses version 7.1.1 – that should explain the difference.

I’ve also updated the taskfile.yml to include descriptions matching the tasks. Additionally, the version in Cargo.toml is now used for the Docker tags. I’ve modified the Dockerfile again and replaced the latest ffmpeg version with the version available in the apk repositories. Now I end up with 194MB, which is only minimally larger than your original one. 

BTW: Thanks a lot for providing this super helpful tool!

tgotwig reviewed (changes requested):
Thanks for your comment and the code changes! Cool that you like the project! 😊

I think we should remove all the docker recipes from the `Taskfile.yml` and only have `run_docker` there.

I would suggest doing it first with the Github Actions and GHCR of your fork with something like `platforms: linux/amd64,linux/arm64` from [here](https://github.com/tgotwig/ubuntu/blob/ffffc57e5f73b1d8b99f449a390903094bcba17f/.github/workflows/docker-build.yml).

hTobi02 commented:
I’ve implemented the changes that were quick and easy for me to address – mainly things that directly affected my workflow. As for the rest (like the GitHub Actions and multi-platform publishing), I’m not too familiar with that, so I’d rather leave it in your hands.

Feel free to keep or discard my changes as you see fit – no hard feelings either way.
